### DISCUSSION TITLE
Generate general type file

### DISCUSSION THREAD

**Discussion by KammererTob** [0no-co/gql.tada#144] [Q&A] [answered]
Hey,

thanks for the great project. I am currently moving over from client-preset and until now it works really great. There is one question, which i was unable to find an answer to: Is there a way to generate types from our graphql schema that can be used in other places of our code? Or is/will this not be the responsibility of this library? In client-preset you could import the types you needed (for example enums) and use them normally in your code (i.e. without any query/mutation).
Here is a more specific example of what i mean:
```vue
        <div class="row">
          <custom-select
            v-model="addition.type"
            :options="Object.values(AdditionType)"
            label="Type"
            class="col input-required"
          />
        </div>
        
        <script setup>
        import { AdditionType } from '@/graphql/generated/graphql'
        ...
        </script>
```
`AdditionType` is an enum and if i for example want the user to be able to select from one of the values. It would be nice to have this automatically generated from the GraphQL schema, so the list is always up-to-date.

I assume one option would be to use the graphql-codegen project and instead use the `typescript` preset instead of the `client-preset` one. From reading the docs this should only generate the types for me.

**Comment by JoviDeCroock** [comment, marked as answer]
You can still import scalar, enum, and input types with the `graphql.scalar` helper https://gql-tada.0no.co/reference/gql-tada-api#graphql-scalar for full-types we encourage folks to use fragment co-location.

**Reply by KammererTob** [reply]
Thanks for your answer. That is indeed helpful for enums. I have a lot of forms for basically nearly complete input types in the GraphQL schema (i.e. create a new item of something), which contain non-nullable properties. What is the best option here? To use `VariablesOf` and then `Partial`?

**Reply by JoviDeCroock** [reply]
`graphql.scalar` also allows inferring input-types and if it's not a literal input-type you can use `VariablesOf` yes

**Reply by KammererTob** [reply]
Perfect. Thank you. I will experiment.

**Reply by KammererTob** [reply]
And another question if i may: Is there a way to change the enum generation to be a TS enum instead of a union? I don't think we can enumerate all possible enum options from a union type, or am i wrong?
i.e. something like this: 
```
type Division = ReturnType<
  typeof graphql.scalar<'Division'>
>;

export const divisionOptions = map(Object.values(Division), (division) => ({
  value: division,
  label: startCase(toLower(division)),
}))
```

**Reply by JoviDeCroock** [reply]
No, we don't return enums as enums are actually also valid JS-Objects so I don't think we can generate those

**Reply by tianyingchun** [reply]
maybe we have to use `grapqhql-codegen` to genereated this?
